hades_logs/parsing.py

```diff
@@ -0,0 +1,103 @@
+from collections import defaultdict, namedtuple
+from datetime import datetime
+from functools import partial
+
+_radius_log_entry = namedtuple(
+    'RadiusLogEntry',
+    ['mac', 'reply', 'groups', 'raw_attributes', 'timestamp']
+)
+
+class ParsingError(ValueError):
+    pass
+
+
+def parse_vlan(vlan):
+    stripped = vlan.strip('"')
+    prefix, name = stripped[:1], stripped[1:]
+    if not name:
+        raise ParsingError("VLAN identifier has no Name: {}".format(name))
+    try:
+        taggedness = {'1': "tagged", '2': "untagged"}[prefix]
+    except KeyError:
+        raise ParsingError("VLAN identifier must start with '1' or '2': {}"
+                           .format(stripped))
+    return "{} ({})".format(name, taggedness)
+
+
+def attrlist_to_dict(attrlist):
+    """Convert a list of (key, value) tuples to a dict of lists"""
+    d = defaultdict(list)
+    for key, val in attrlist:
+        d[key].append(val)
+    return d
+
+
+class RadiusLogEntry(_radius_log_entry):
+    """Class representing a parsed hades log entry.
+
+    This is namedtuple subclass conforming with the tuples returned by
+    ``get_auth_attempts_at_port`` from the Hades RPC API.
+
+    It provides convenient access to whether the request was accepted
+    (:py:meth:`__bool__`), the :py:attr:`vlans`, time, and equality.
+    """
+    @property
+    def accepted(self):
+        """Return whether the reply says ``"Access-Accept"``"""
+        return self.reply == "Access-Accept"
+
+    @property
+    def attributes(self):
+        """The attributes provided as a dict of lists."""
+        return attrlist_to_dict(self.raw_attributes)
```

Comment:
It's probably better to either eagerly convert the list of attribute-value-pairs in `__new__` or cache the property with [`functools.lru_cache`](https://docs.python.org/3/library/functools.html#functools.lru_cache), because this property is needed multiple times and its always needed.

I'd go for the eager method.